.github/workflows/ci.yml

```diff
@@ -0,0 +1,155 @@
+name: CI
+on:
+  push:
+  pull_request:
+
+jobs:
+
+  dependencies:
+    name: Prepare Dependencies
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '1.17'
+      - name: Setup dependencies
+        uses: ./.github/actions/deps-setup
+
+  detect-repo-changes:
+    name: Detected Repo Changes
+    runs-on: ubuntu-20.04
+    outputs:
+      helms-changed: ${{ steps.changed_charts.outputs.matrix }}
+      no-changes: ${{ steps.changed_charts.outputs.no_changes}}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - uses: dorny/paths-filter@v2.9.1
+        id: filter
+        with:
+          base: master
+          # `-helm` suffix aims that we can easily filter helm charts
+          filters: |
+            hydra-helm:
+              - 'helm/charts/hydra/**'
+            hydra-maester-helm:
+              - 'helm/charts/hydra-maester/**'
+            keto-helm:
+              - 'helm/charts/keto/**'
+            kratos-helm:
+              - 'helm/charts/kratos/**'
+            kratos-selfservice-ui-node-helm:
+              - 'helm/charts/kratos-selfservice-ui-node/**'
+            oathkeeper-helm:
+              - 'helm/charts/oathkeeper/**'
+            oathkeeper-maester-helm:
+              - 'helm/charts/oathkeeper-maester/**'
+      # This step will take output from paths-filter and then process it in order to get what helm charts have been updated
+      # It allows us to do matrix in validating & testing process
+      - name: Generate helm chart matrix to be validated
+        id: changed_charts
+        env:
+          FILTER_OUTPUT: ${{ toJson(steps.filter.outputs) }}
+        run: |
+           updated_charts=$(echo "$FILTER_OUTPUT" | jq -r 'to_entries | map(select((.key | endswith("-helm")) and .value == "true")) | map(.key)')
+           updated_charts=$(echo "$updated_charts" | sed "s/-helm//g")
+
+           if [[ "$updated_charts" == "[]" ]]; then 
+            echo ::set-output name=no_changes::"true"
+           fi
+           echo ::set-output name=matrix::{\"chart\":$(echo $updated_charts)}\"
+           
+           echo "Charts array to run CI on: $updated_charts"
+  
+  check:
+    name: Check Helm Chart '${{ matrix.chart }}'
+    if: ${{ needs.detect-repo-changes.outputs.no-changes != 'true' }}
```

Comment:
This line prevents crashes when none helm chart has been updated. It also prevents release to run when no chart updated